app/scripts/lib/app-start.js

```diff
@@ -588,6 +564,51 @@ define(function (require, exports, module) {
       }
     },
 
+    /**
+     * Check whether there are any problems accessing localStorage.
+     * Errors are logged to Sentry and internal metrics.
+     *
+     * If there is a problem accessing localStorage, the user
+     * will be redirected to `/cookies_disabled` from _selectStartPage
+     */
+    testLocalStorage: function () {
+      var self = this;
+      return p().then(function () {
+        // only test localStorage if the user is not already at
+        // the cookies_disabled screen.
+        if (! self._isAtCookiesDisabled()) {
+          self._storage.testLocalStorage(self._window);
+        }
+      }).fail(self.captureError.bind(self));
```

Comment:
I tried that originally. I ditched [_selectStartPage](https://github.com/mozilla/fxa-content-server/pull/3424/files#diff-b80f22cfd726dfdc2bb7ed3e82dc7264R797) and instead threw an error if localStorage was disabled. In `_getErrorPage`, I added a new branch to see if the error came from the storage module. If so, redirect to `cookies_disabled`, if the user was not already at cookies disabled. Since it used a redirect instead of pushState, it was a bit slower, and also broke w/ x-frame-options headers - [/cookies_disabled is not on the list of allowed pages](https://github.com/mozilla/fxa-content-server/blob/35abdf1705471b91614ac94858d364899154a6fb/server/lib/routes.js#L127).